iris_grib/_load_convert.py

```diff
@@ -848,17 +888,30 @@ def _calculate_proj_coords_from_lon_lat(section, cs):
     # but the distance measurement is in 10-3 m, so a conversion is necessary
     # to find the origin in m.
 
+    if section['gridDefinitionTemplateNumber'] in _XY_GDT_NUMBERS:
+        dx = section['Dx']
+        dy = section['Dy']
+        nx = section['Nx']
+        ny = section['Ny']
+    elif section['gridDefinitionTemplateNumber'] in _IJ_GDT_NUMBERS:
+        dx = section['Di']
+        dy = section['Dj']
+        nx = section['Ni']
+        ny = section['Nj']
+    else:
+        raise TranslationError('Unsupported lat-lon point parameters')
+
     scan = scanning_mode(section['scanningMode'])
     lon_0 = section['longitudeOfFirstGridPoint'] * _GRID_ACCURACY_IN_DEGREES
     lat_0 = section['latitudeOfFirstGridPoint'] * _GRID_ACCURACY_IN_DEGREES
     x0_m, y0_m = cs.as_cartopy_crs().transform_point(
         lon_0, lat_0, ccrs.Geodetic())
-    dx_m = section['Dx'] * 1e-3
-    dy_m = section['Dy'] * 1e-3
+    dx_m = dx * 1e-3
```

Comment:
Are you sure this constant offset applies equally to i/j as well as to x/y?

Reply:
You could add some testing to confirm whether it does or not 😄

Reply:
I thought it did but having read the ![template definitions](http://www.wmo.int/pages/prog/www/WMOCodes/WMO306_vI2/LatestVERSION/WMO306_vI2_GRIB2_Template_en.pdf) more closely the units of Di/Dj/Dx/Dy can have units of 10^-2 m, 10^-3 m, 10^-6 degrees or None :worried:

Reply:
I will refactor it to only use grid length (which are still either `Di`+`Dj` or `Dx`+`Dy`). These all use 10^-3 m as their unit. I have renamed things accordingly.